faculty/clients/job.py

```diff
@@ -256,9 +387,47 @@ def list(self, project_id):
         endpoint = "/project/{}/job".format(project_id)
         return self._get(endpoint, JobSummarySchema(many=True))
 
-    def update_metadata(self, project_id, job_id, name, description):
+    def create_job(self, project_id, job_parameters):
+        """Create a job.
+
+        Parameters
+        ----------
+        project_id : uuid.UUID
+        job_parameters : Dict[dict]
```

Comment:
Users of this API will have to know how to construct a fairly complicated nested dictionary. Can we instead have an API like:

```python
def create(
    self,
    project_id,
    name,
    description,
    definition
):
    # etc.
```

Then the user can pass in a `JobDefinition` object, which is harder to get wrong (though admittedly still non-trivial).

Note that I've deliberately not used the `JobMetadata` object as an argument, because:
* We can't set the `author_id`, `created_at` or `last_updated_at` so would need a new version of that object with those attributes omitted.
* There are only two keys remaining, and it seems a bit over the top to force the user to put them in an object first.
* It also mirrors the signature of `update_metadata`.

We can then use something like the following to create the payload:

```python
schema = JobDefinitionSchema()
payload = {
    "meta": {
        "name": name,
        "description": description
    },
    "definition": schema.dump(definition)
}
```